mdnsd/kiface.c

```diff
@@ -54,6 +54,10 @@ struct {
 } kev_state;
 
 
+#ifdef __FreeBSD__
+#define rtm_hdrlen rtm_msglen
+#endif
```

Comment:
I suspect this is wrong -- on OpenBSD rtm_hdrlen is documented as being
```
	u_short	rtm_hdrlen;	/* sizeof(rt_msghdr) to skip over the header */
```
while rtm_msglen on both FreeBSD and OpenBSD is documented as being
```
	u_short	rtm_msglen;	/* to skip over non-understood messages */
```
so at kiface.c:150
```
                sa = (struct sockaddr *)(buf + rtm->rtm_hdrlen);
                get_rtaddrs(rtm->rtm_addrs, sa, rti_info);
```
you'll be skipping the whole message rather than just the header of the message.